interfaces/devdoc/file_util_linux.md

```diff
@@ -0,0 +1,67 @@
+# file_util_linux
+
+## Overview
+
+Linux implementation of the file operation functions.
+
+## Exposed API 
+
+```c
+// Most defines removed for space constraints
+#define INVALID_HANDLE_VALUE            NULL
+
+typedef void *HANDLE;
+typedef void *LPSECURITY_ATTRIBUTES;
+
+MOCKABLE_FUNCTION(, HANDLE, ebs_create_file, const char*, lpFileName, unsigned long, dwDesiredAccess, unsigned long, dwShareMode, LPSECURITY_ATTRIBUTES, lpSecurityAttributes, unsigned long, dwCreationDisposition, unsigned long, dwFlagsAndAttributes, HANDLE, hTemplateFile);
+MOCKABLE_FUNCTION(, bool, ebs_close_handle, HANDLE, handle_input);
+```
+
+### ebs_create_file
+
+```c
+MOCKABLE_FUNCTION(, HANDLE, ebs_create_file, const char*, lpFileName, unsigned long, dwDesiredAccess, unsigned long, dwShareMode, LPSECURITY_ATTRIBUTES, lpSecurityAttributes, unsigned long, dwCreationDisposition, unsigned long, dwFlagsAndAttributes, HANDLE, hTemplateFile);
+```
+
+`ebs_create_file` implements the creating file function in Linux. Uses the `open` function in C for Linux.
+
+**SRS_FILE_UTIL_LINUX_09_001: [** If the `full_file_name` input is either empty or `NULL`, `ebs_create_file` shall return an `INVALID_HANDLE_VALUE`. **]**
+
+**SRS_FILE_UTIL_LINUX_09_002: [** `ebs_create_file` shall allocate memory for the file handle. ebs_create_file will succeed and return a non-`NULL` value.  **]**
+
+**SRS_FILE_UTIL_LINUX_09_003: [** If memory allocation for `result` fails, `ebs_create_file` shall return an `INVALID_HANDLE_VALUE`. **]**
+
+**SRS_FILE_UTIL_LINUX_09_004: [** If `dwDesiredAccess` is `GENERIC_READ`, `ebs_create_file` will call `open` with `O_RDONLY` and shall return a file handle for read only.  **]**
+
+**SRS_FILE_UTIL_LINUX_09_005: [** If `dwDesiredAcces` is `GENERIC_WRITE`, `ebs_create_file` will call `open` with `O_WRONLY` and shall return a file handle for write only.  **]**
+
+**SRS_FILE_UTIL_LINUX_09_006: [** If `dwDesiredAccess` is `GENERIC_ALL` or `GENERIC_READ&GENERIC_WRITE`, `ebs_create_file` will call `open` with `O_RDWR` and shall return a file handle for read and write.  **]**
+
+**SRS_FILE_UTIL_LINUX_09_014: [** If `creation_disposition` is `CREATE_ALWAYS` or `OPEN_ALWAYS`, `ebs_create_file` will call `open` with `O_CREAT` and shall either create a new file handle if the specificied pathname exists and return it or return an existing file handle.  **]**
```

Comment:
>creation_disposition [](http://example.com/codeflow?start=38&length=20)

Doesn't match parameter name (but maybe the parameter should be changed to this anyway)